design/cstor/v1/cstorpool.md

```diff
@@ -247,9 +247,28 @@ type CStorPoolInstanceStatus struct {
 
 // CStorPoolInstanceCapacity stores the pool capacity related attributes.
 type CStorPoolInstanceCapacity struct {
+	// Amount of physical data (and its metadata) written to pool
+	// after applying compression, raid properties, etc..,
+	Used resource.Quantity `json:"used"`
+	// Amount of usable space in the pool after excluding
+	// metadata and raid properties
+	Free resource.Quantity `json:"free"`
+	// Sum of usable capacity in all the data raidgroups
 	Total resource.Quantity `json:"total"`
-	Free  resource.Quantity `json:"free"`
-	Used  resource.Quantity `json:"used"`
+	// ZFSCapacityAttributes contains advanced information about pool capacity details
+	ZFS ZFSCapacityAttributes `json:"zfs"`
```

Comment:
Simply `ZFS` as a key does not make sense as by seeing the key one cannot comment on what it's purpose is!
It is too generic IMO. 
Also it makes sense like following 
```
ZFSCapacityAttributes ZFSCapacityAttributes `json:"ZFSCapacityAttributes"` 
```
I mean we can have key and its type as same string in order to be more specific and vivid.

e.g https://github.com/kubernetes/api/blob/master/core/v1/types.go#L322

Now the problem with `ZFSCapacityAttributes` as a key is -- 
Are `Total` , `Used` and `Free` not ZFS attributes ? If I am not wrong -- those are too ZFS attributes. I think we are dealing with ZFS attributes only here.

Reply:
Yes!! We are dealing with ZFS attributes.

Reply:
One of the thing that I think I can propose is : 
We segregated `ZFSCapacityAttributes` with intentions to be more clear on what user needs to focus and maybe not worried about complex things inside `ZFSCapacityAttributes` but still, that catches the eye.

e.g https://github.com/kubernetes/api/blob/master/apps/v1/types.go#L376
I can still see some fields inside the structure that I don't care and only make sense with what is useful to me right now. 

My point is -- to solve this naming problem --  we can have all the capacity attributes inside this capacity structure.

Reply:
Open to discussions and above are my nit comments only. cc @kmova @mittachaitu